samples/drivers/led_ws2812/README.rst

```diff
@@ -0,0 +1,129 @@
+#. Connect the 5V power supply pin to the 5V input of the LED strip.
+
+Building and Running
+********************
+
+.. _blog post on WS2812 timing: https://wp.josh.com/2014/05/13/ws2812-neopixels-are-not-so-finicky-once-you-get-to-know-them/
+
+The sample application is located at ``samples/drivers/led_ws2812/``
+in the Zephyr source tree.
+
+Configure For Your LED Strip
+----------------------------
+
+The first thing you need to do is make sure that the driver is
+configured to match the particular LED chips you're using. For
+example, some chips have a "green, red, blue" color ordering on their
+data pin, while others have "red, green, blue, white". Check your
+chip's datasheet for your configuration (though given the number of
+competing implementations and knock-offs, some trial and error may be
+necessary).
+
+To make sure the driver is set up properly for your chips, check the
+values of the following configuration options:
+
+- ``CONFIG_WS2812_RED_ORDER``
+- ``CONFIG_WS2812_GRN_ORDER``
+- ``CONFIG_WS2812_BLU_ORDER``
+- ``CONFIG_WS2812_WHT_ORDER`` (available if your chips have a white
+  channel by setting ``CONFIG_WS2812_HAS_WHITE_CHANNEL`` to ``y``).
+
+Refer to their help strings for details.
+
+Configure For Your Board
+------------------------
+
+Now check if your board is already supported, by looking for a file
+named ``boards/YOUR_BOARD_NAME.conf`` in the application directory.
+
+If your board isn't supported yet, you'll need to configure the
+application as follows.
+
+#. Configure your board's SPI master in a configuration file under
+   ``boards/`` in the sample directory.
+
+   To provide additional configuration for some particular board,
+   create a ``boards/YOUR_BOARD_NAME.conf`` file in the application
+   directory. It will be merged into the application configuration.
+
+   In this file, you must ensure that the SPI peripheral you want to
+   use for this demo is enabled, and that its name is "ws2812_spi".
+   See ``boards/96b_carbon.conf`` for an example, and refer to your
+   board's configuration options to set up your desired SPI master.
+
+#. Set the number of LEDs in your strip in the application sources.
+   This is determined by the macro ``STRIP_NUM_LEDS`` in the file
+   ``src/main.c``. The value in the file was chosen to work with the
+   AdaFruit NeoPixel Ring 12.
+
+#. Make sure that the SPI timing will generate pulses of the
+   appropriate widths. This involves setting the configuration options
+   ``CONFIG_WS2812_STRIP_SPI_BAUD_RATE``,
+   ``CONFIG_WS2812_STRIP_ONE_FRAME``, and
+   ``CONFIG_WS2812_STRIP_ZERO_FRAME``. Refer to the Kconfig help for
+   those options, as well as this `blog post on WS2812 timing`_, for
+   more details.
+
+Then build and flash the application:
+
+.. code-block:: bash
+
+   $ cd ~/zephyr
+   $ source zephyr-env.sh
+   $ cd samples/drivers/led_ws2812
+   $ BOARD=your_board make flash
+
+Refer to your :ref:`board's documentation <boards>` for alternative
```

Comment:
the ``:ref:`name` `` notation requires the specific `name` to be defined in some document. So for here you should write:

     Refer to your board's documentation for alternative

Reply:
I'm not sure if GitHub ate the formatting (I don't see `:ref:` anywhere in the above comment), but `boards` is a valid ref target when I build the documentation locally. It points here:

https://raw.githubusercontent.com/zephyrproject-rtos/zephyr/master/boards/boards.rst

The extended syntax of `:ref:` in this PR allows setting the link text and its ref target separately. I did it that way because "refer to your board's documentation" seemed to flow better than "refer to Supported Boards", but I'm happy to change it if that'd be better. Please let me know.

Reply:
doh, your're right.  As you wrote it is just right, so ignore my comment on this fix.